module/os/freebsd/zfs/zfs_file_os.c

```diff
@@ -239,7 +239,7 @@ zfs_file_fsync(zfs_file_t *fp, int flags)
 	if (fp->f_type != DTYPE_VNODE)
 		return (EINVAL);
 
-	v = fp->f_data;
+	v = fp->f_vnode;
 	return (zfs_vop_fsync(v));
```

Comment:
Can we do away with the local variable entirely.

```suggestion
	return (zfs_vop_fsync(fp->f_vnode));
```